### PR TITLE
Validate attachments

### PR DESCRIPTION
**Pull request by mkangia**
## Product Description

Currently there is a validation on attachments with the form on [CommCare app](https://github.com/dimagi/commcare-android/blob/ef45074bbb8647339387e179a2baca89fb394a23/app/src/org/commcare/utils/FormUploadUtil.java#LL63C1-L63C1) and also on [Webapps](https://github.com/dimagi/commcare-hq/pull/32634/files#diff-4f03cd5cfcedb9050fc1aaaac032fd594ebfecb173feecdfecfcf2fc9f6902a9R914).
This change adds the same validation as on formplayer on CommCareHQ as well. 
Since a form submission can come from ways other than CommCare app or Webapps, a check on HQ is essential.
The motivation for the change came from an audit where they considered the [OWASP recommendations on file upload](https://owasp.org/www-community/vulnerabilities/Unrestricted_File_Upload).

## Technical Summary

There are 2 things that are checked, file extension and content type. 
The logic in HQ has been kept same as [that on webapps](https://github.com/dimagi/commcare-hq/pull/32634/files#diff-4f03cd5cfcedb9050fc1aaaac032fd594ebfecb173feecdfecfcf2fc9f6902a9R960) i.e content_type (equivalent of mimetype) starts with `image/` or `audio/` or `video/` and extensions as permitted on webapps.

### Safety story

Tests added. Tested locally that normal form submission is unaffected. Will take this through QA as well.

### QA Plan

https://dimagi-dev.atlassian.net/browse/QA-5214

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by mkangia** [comment]
Thanks for taking a look @Robert-Costello 

> The webapps validation was pretty finicky.

Can you tell more?

**Comment by Robert-Costello** [comment]
> Can you tell more?

The one I was thinking of most was the difference in how windows machines and macs behaved. For example, if the input was for an image type you would only be able to select a file that was some kind of image on a mac (same for audio and video). On windows machines you were allowed to choose incorrect file types, video file for image questions etc. So that issue called for more strict checking on the frontend. Mapping extensions to mime type and matching that to the input.

**Comment by mkangia** [comment]
Cool, thanks @Robert-Costello

File: corehq/ex-submodules/couchforms/getters.py
Comment (snopoke) [inline corehq/ex-submodules/couchforms/getters.py:90]: This code is almost identical to `_valid_instance_file_extension`. I think they could be combined and just pass in the list of acceptable extensions.

File: corehq/ex-submodules/couchforms/getters.py
Comment (snopoke) [inline corehq/ex-submodules/couchforms/getters.py:97]: Formplayer also allows `application/pdf`

Comment (mkangia) [reply corehq/ex-submodules/couchforms/getters.py:97]: Ah, bad miss.
Thanks for catching that @snopoke

File: corehq/ex-submodules/couchforms/getters.py
Comment (mkangia) [reply corehq/ex-submodules/couchforms/getters.py]: Yes absolutely.

https://github.com/dimagi/commcare-hq/pull/32998/commits/21c4e89d4e196875723b631fed95e65a644ee048

**Comment by mkangia** [comment]
Thanks for the review @snopoke.
This is ready for a second pass.

File: corehq/apps/receiverwrapper/tests/test_submissions.py
Comment (mkangia) [reply corehq/apps/receiverwrapper/tests/test_submissions.py:150]: Hey @mjriley 

> I prefer to see the entire expectation in the test name

For tests, I would usually just name what is being tested. The expectations would be in the test. I see other tests in this file follow the same naming pattern.

> Is the name here implying that 'application/octet-stream' is a format exclusive to mobile, or that the content type is the only part of this that is valid (as the extension isn't?)

the name is implying the former. Invalid extension was taken for the test, to test that the format is accepted irrespective of valid/invalid extension.

> It might be helpful to replace it with a constant

cool.

File: corehq/apps/receiverwrapper/tests/test_submissions.py
Comment (mkangia) [reply corehq/apps/receiverwrapper/tests/test_submissions.py:191]: So this is what we are sending back to the app. The XML syntax used here is what CommCare mobile app and HQ use to communicate. The app will show just the message only to the user.

File: corehq/ex-submodules/couchforms/getters.py
Comment (mjriley) [reply corehq/ex-submodules/couchforms/getters.py:98]: I'm fine with going forward with this PR, but I want to point out the downsides to this approach. It is generally much easier to focus on the edge cases when testing at a low level. When we test this same behavior at a high level, the tests become harder to read, take longer to execute, and, at least from what I've seen...stand a much higher chance of never getting written.

Comment (mkangia) [reply corehq/ex-submodules/couchforms/getters.py:98]: thanks @mjriley 
Please drop an approval if you are comfortable approving the PR. I am waiting on an approval to merge this. It is ready to go with QA done.

**Review by mjriley — APPROVED** [review]

**Review by snopoke — APPROVED** [review]

**Pull request merged**
